File: examples/minimal_api_example.py
#!/usr/bin/env python3
# Minimal example of using the Deep Research API with Python

import requests

# Configuration
API_URL = "https://your-deployment-url.com/api/research/query"
API_KEY = "your-access-password"
RESEARCH_QUERY = "History of cars"

# Function to perform research
def perform_research():
    try:
        print(f'Researching: "{RESEARCH_QUERY}"...')
        
        headers = {
            "Content-Type": "application/json",
            "Authorization": f"Bearer {API_KEY}"
        }
        
        data = {
            "query": RESEARCH_QUERY
        }
        
        response = requests.post(API_URL, json=data, headers=headers)
        response.raise_for_status()  # Raise exception for HTTP errors
        
        result = response.json()
        print("\n=== RESEARCH REPORT ===\n")
        print(result["report"])
        
    except Exception as e:
        print(f"Error: {e}")

# Run the research
if __name__ == "__main__":
    perform_research()

"""
To use this script:
1. Replace API_URL with your actual deployment URL
2. Replace API_KEY with your access password
3. Replace RESEARCH_QUERY with your research topic
4. Install requests: pip install requests
5. Run the script: python minimal_api_example.py
"""

File: examples/minimal-api-example.js
// Minimal example of using the Deep Research API
const fetch = require('node-fetch');

// Configuration
const API_URL = 'https://your-deployment-url.com/api/research/query';
const API_KEY = 'your-access-password';
const RESEARCH_QUERY = 'History of cars';

// Function to perform research
async function performResearch() {
  try {
    console.log(`Researching: "${RESEARCH_QUERY}"...`);
    
    const response = await fetch(API_URL, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        'Authorization': `Bearer ${API_KEY}`
      },
      body: JSON.stringify({ query: RESEARCH_QUERY })
    });
    
    if (!response.ok) {
      throw new Error(`API error: ${response.status}`);
    }
    
    const result = await response.json();
    console.log('\n=== RESEARCH REPORT ===\n');
    console.log(result.report);
    
  } catch (error) {
    console.error('Error:', error.message);
  }
}

// Run the research
performResearch();

/*
To use this script:
1. Replace API_URL with your actual deployment URL
2. Replace API_KEY with your access password
3. Replace RESEARCH_QUERY with your research topic
4. Install node-fetch: npm install node-fetch
5. Run the script: node minimal-api-example.js
*/
